### PR TITLE
Create a tutorial notebook for Quantum Entanglement

### PR DESCRIPTION
**Pull request by jyotiraj-code**
Hi @amir-naveh, this current pull request closes the #28 issue of creating a tutorial notebook for the quantum entanglement. Included in the notebook is:

- A comprehensive hands on tutorial for quantum entanglement taking the structure reference from the [superposition notebook](https://github.com/Classiq/classiq-library/tree/main/community/basic_examples/superposition)
- Created proper mathematical comprehension with the algebra to make sure the tutorial is self sufficient
- Used proper methods to describe the parameters of the Jupyter notebook and generated the `.qmod` and `.json` file

Necessary instructions have been followed to create the pull request and confirmed there are no typos in the changes made

_This particular PR is issued as a part of the UnitaryHack 2024 event_

### CODE REVIEW

File: community/basic_examples/entanglement/entanglement.ipynb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    "0 & 0 & 1 & 0\n",
+    "\\end{pmatrix}$$\n",
+    "The new quantum state is:\n",
+    "$$\\begin{pmatrix} 1 & 0 & 0 & 0 \\\\ 0 & 1 & 0 & 0 \\\\ 0 & 0 & 0 & 1 \\\\ 0 & 0 & 1 & 0 \\end{pmatrix} \\begin{pmatrix} \\frac{1}{\\sqrt{2}} \\\\ 0 \\\\ \\frac{1}{\\sqrt{2}}\\\\ 0 \\end{pmatrix} = \\begin{pmatrix} \\frac{1}{\\sqrt{2}} \\\\ 0 \\\\ 0 \\\\ \\frac{1}{\\sqrt{2}} \\end{pmatrix} = \\frac{1}{\\sqrt{2}} \\left| 00 \\right\\rangle + \\frac{1}{\\sqrt{2}} \\left| 11 \\right\\rangle$$\n",
+    "\n",
+    "In this situation the qubit's of Alice and Bob are correlated to each other.<br>\n",
+    "If we measure the both the qubits we will either get the state $|00\\rangle$ or the state $|11\\rangle$ with equal probability.\n",
+    "\n",
+    "It is described as:\n",
+    "- When the Alice oberseve her state as $|0\\rangle$ then the state of Bob's qubit collapses to the state $|0\\rangle$\n",
```
Comment (amir-naveh) [inline community/basic_examples/entanglement/entanglement.ipynb:80]: When the Alice oberseve  -> When Alice observes

File: community/basic_examples/entanglement/entanglement.ipynb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    "- When the Alice oberseve her state as $|1\\rangle$ then the state of Bob's qubit collapses to the state $|1\\rangle$ \n",
```
Comment (amir-naveh) [inline community/basic_examples/entanglement/entanglement.ipynb:81]: When the Alice oberseve -> When Alice observes

Comment (amir-naveh) [reply community/basic_examples/entanglement/entanglement.ipynb:81]: Hey @jyotiraj-code - great work. Two small typos. If you could commit the changes I will approve and merge. Thank you!

**Comment by jyotiraj-code** [comment]
Hi @amir-naveh made the changes in three(?) typos. If any more changes need to be made please let me know, I would be happy to do so

**Review by amir-naveh — APPROVED** [review]

**Comment by amir-naveh** [comment]
Merged and approved. Thank you @jyotiraj-code !

**Pull request merged**